ReSharper.FSharp/src/FSharp.Psi.Features/src/Parsing/FSharpImplTreeBuilder.fs

```diff
@@ -624,10 +630,12 @@ type FSharpImplTreeBuilder(lexer, document, decls, lifetime, path, projectedOffs
             if Position.posGt r.End outerRange.Start then attrs else
             x.SkipOuterAttrs(rest, outerRange)
 
-    member x.ProcessTopLevelBinding(binding) =
-        let (SynBinding(_, kind, _, _, attrs, _, _ , headPat, returnInfo, expr, _, _, _)) = binding
+    member x.ProcessTopLevelBinding(binding, isSecondary) =
+        let (SynBinding(_, kind, _, _, attrs, XmlDoc xmlDoc, _ , headPat, returnInfo, expr, range, _, _)) = binding
 
-        let mark = x.Mark()
+        if isSecondary then x.AdvanceToTokenOrRangeStart(FSharpTokenType.AND, range)
```

Comment:
Do these cases work correctly?
```
let rec f x = ()

/// Yo
and g x = ()
```

```
let rec f x = ()

and g x = ()
```

If yes, is checking the token type needed, or checking the range is sufficient?

Reply:
> Do these cases work correctly?

Yes, there are tests for that:

`Let - Rec 02 - And `
`Let - Top 03 - XmlDoc 01`

> If yes, is checking the token type needed, or checking the range is sufficient?

Yes, still required since `and` is not in binding range

Reply:
Hm, it looks like I've been looking at `debugPoint` field when writing the comment: its range does include the keyword but I haven't checked if we can use this field reliably.

https://fsprojects.github.io/fantomas-tools/#/ast?data=N4KABGBEAmCmBmBLAdrAzpAXFSAacUiaAYmolmPAIYA2as%2BEkAxgPZwWQ2wAuYATrGaUwADzABeMAAoAlAB1kiqsmhgA5mMkzZkEAF8gA

https://fsprojects.github.io/fantomas-tools/#/ast?data=N4KABGBEAmCmBmBLAdrAzpAXFSAacUiaAYmolmPAIYA2as%2BEkAxgPZwWQ2wAuYATrGaUwADzABeMAAoAlAB1kigPSqwATVaKqyaGADmYyTNmQQAXyA